xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp

```diff
@@ -531,9 +532,26 @@ bool CDVDVideoCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptio
 
   if (m_render_surface)
   {
-    m_jnivideosurface = CXBMCApp::get()->getVideoViewSurface();
+    m_jnivideoview.reset(CJNIXBMCVideoView::createVideoView(this));
+    if (!m_jnivideoview || !m_jnivideoview->waitForSurface(500))
```

Comment:
I'm not 100% sure what the `!m_jnivideoview->waitForSurface(500)` part does. It should only return true for when the 500ms are reached. So it might return false for 0-499 and 501-XX. Our documentations isn't explaining that very well.